Expose more internal dataset information as fields

jeremyh opened the pull request:
There's a lot of information, such as the creation time of datasets, that we use regularly but is very inconsistently made visible across our tools and UIs.

This adds four fields to expose more internal information:
- `indexed_time` (when the dataset was indexed)
- `indexed_by` (user who indexed the dataset)
- `creation_time` (creation of dataset: when it was processed)
- `label` the label for a dataset

The first three are already recorded on all datasets, the last is optional (and is defined in GA's scenes, and some others).

Adding them as fields means they will be:
- Shown on most UIs: (`datacube dataset search|show`,  cubedash screens)
- selectable from skinny searches: `index.datasets.search_returning(('id', 'creation_time'), product='ls7_level1_scene, ...)`
- Filterable: `index.datasets.search(product="..", indexed_by="dg6911")`

(for the document-based ones:)
- Available on Dataset objects: `dataset.metadata.creation_time`

jeremyh committed:
Expose more fields: creation/indexing times, labels, users

jeremyh committed:
Add new fields to test cases

omad reviewed (changes requested):
This looks great Jeremy. Nice explanation in the PR!

Can you please add an item into the `whats_new.rst` and I'll merge it in. A short summary with a link to the PR.

I want better documentation around all of the metadata_type/dataset/searching features. But we don't even have a structure in place for adding it. This is a worthwhile change anyway that I won't hold it up pending such a mountain of a task. But in future if you have time, can you at least make a start on including some user docs.

Cheers

jeremyh committed:
Add filter test for new fields

jeremyh committed:
Add limit option to search_returning()

It should match search()

jeremyh committed:
Export 'format' as a field too

jeremyh committed:
Re-add default values for fixed fields

These used to be optional on metadata types, but it looks
like it was lost during the unification refactor.

The default telemetry type is missing 'format', so throws
an error without the default.

jeremyh committed:
Consistency of field labels

omad committed:
Update `whats_new` documentation
